crates/core/tedge_agent/src/lib.rs

```diff
@@ -53,20 +53,19 @@ pub struct AgentOpt {
 }
 
 pub async fn run(agent_opt: AgentOpt) -> Result<(), anyhow::Error> {
-    let tedge_config_location =
-        tedge_config::TEdgeConfigLocation::from_custom_root(agent_opt.common.config_dir.clone());
-
     log_init(
         "tedge-agent",
         &agent_opt.common.log_args,
-        &tedge_config_location.tedge_config_root_path,
+        &agent_opt.common.config_dir,
     )?;
 
+    let tedge_config = tedge_config::TEdgeConfig::load(&agent_opt.common.config_dir).await?;
```

Comment:
The config is read a second time here. It could be passed as an argument to `run()`.

Reply:
Fixed: https://github.com/thin-edge/thin-edge.io/pull/3625/commits/3d32c78b31d78607dcf9854075f885b48006abd6